src/main/java/com/wolox/training/model/Book.java

```diff
@@ -0,0 +1,144 @@
+package com.wolox.training.model;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+@Entity
+public final class Book {
+
+    private @Id
+    Long idBook;
+    private @Column
+    String genre;
+    private @Column(nullable = false)
+    String author;
+    private @Column(nullable = false)
+    String image;
+    private @Column(nullable = false)
+    String title;
+    private @Column(nullable = false)
+    String subtitle;
+    private @Column(nullable = false)
+    String publisher;
+    private @Column(nullable = false)
+    String year;
+    private @Column(nullable = false)
+    Integer pages;
+    private @Column(nullable = false)
+    String isbn;
+
+    Book(){}
+
+    public Book(Long idBook, String genre, String author, String image, String title,
+        String subtitle, String publisher, String year, Integer pages, String isbn) {
+        this.idBook = idBook;
+        this.genre = genre;
+        this.author = author;
+        this.image = image;
+        this.title = title;
+        this.subtitle = subtitle;
+        this.publisher = publisher;
+        this.year = year;
+        this.pages = pages;
+        this.isbn = isbn;
+    }
+
+    public void setIdBook(Long idBook) {
```

Comment:
Remove this method, JPA takes care of giving the values to the id.
We don't want that the id of an object can be changed, that the only one that can modify it is JPA.